### PR TITLE
Add opaque secret creation 2.7 stolostron/backlog#27002

### PR DESCRIPTION
**Pull request by oafischer**
https://github.com/stolostron/backlog/issues/27002

### CODE REVIEW

**Review by swopebe — APPROVED** [review]
Maybe make clearer that it is the one thing that cannot be done in the console.

File: multicluster_engine/credentials/credential_vm.adoc [hunk 1]
```diff
@@ -106,3 +106,18 @@ You can create a cluster that uses this credential by completing the steps in xr
...
+[#vsphere-create-opaque-secret-api]
+== Creating an opaque secret by using the API
+
+To create an opaque secret for VMware vSphere by using the API, apply YAML content that is similar to the following example:
+
+[source,yaml]
+----
+kind: Secret
+metadata:
+    name: <managed-cluster-name>-vsphere-creds
```
Comment (oafischer) [inline multicluster_engine/credentials/credential_vm.adoc]: Let's double check that the provider short name is indeed vsphere

File: multicluster_engine/credentials/credential_aws.adoc [hunk 2]
```diff
@@ -51,3 +51,18 @@ You can edit your credential in the console. If the cluster was created by using
...
+To create an opaque secret for Amazon Web Services by using the API, apply YAML content that is similar to the following example:
+
+[source,yaml]
+----
+kind: Secret
+metadata:
+    name: <managed-cluster-name>-aws-creds
+    namespace: <managed-cluster-namespace>
+type: Opaque
+data:
```
Comment (manosnoam) [inline multicluster_engine/credentials/credential_aws.adoc]: The data part is missing, it is unique to AWS platform.

File: multicluster_engine/credentials/credential_azure.adoc [hunk 3]
```diff
@@ -70,3 +70,18 @@ You can create a cluster that uses this credential by completing the steps in xr
...
+To create an opaque secret for Microsoft Azure by using the API, apply YAML content that is similar to the following example:
+
+[source,yaml]
+----
+kind: Secret
+metadata:
+    name: <managed-cluster-name>-azure-creds
+    namespace: <managed-cluster-namespace>
+type: Opaque
+data:
```
Comment (manosnoam) [inline multicluster_engine/credentials/credential_azure.adoc]: The data part is missing, it is unique to Azure platform.

File: multicluster_engine/credentials/credential_google.adoc [hunk 4]
```diff
@@ -52,3 +52,18 @@ You can use this connection when you create a cluster by completing the steps in
...
+To create an opaque secret for Google Cloud Platform by using the API, apply YAML content that is similar to the following example:
+
+[source,yaml]
+----
+kind: Secret
+metadata:
+    name: <managed-cluster-name>-gcp-creds
+    namespace: <managed-cluster-namespace>
+type: Opaque
+data:
```
Comment (manosnoam) [inline multicluster_engine/credentials/credential_google.adoc]: The data part is missing, it is unique to GCP platform.

File: multicluster_engine/credentials/credential_openstack.adoc [hunk 5]
```diff
@@ -89,3 +89,18 @@ You can create a cluster that uses this credential by completing the steps in xr
...
+To create an opaque secret for Red Hat OpenStack Platform by using the API, apply YAML content that is similar to the following example:
+
+[source,yaml]
+----
+kind: Secret
+metadata:
+    name: <managed-cluster-name>-osp-creds
+    namespace: <managed-cluster-namespace>
+type: Opaque
+data:
```
Comment (manosnoam) [inline multicluster_engine/credentials/credential_openstack.adoc]: The data part is missing, it is unique to Openstack platform.

File: multicluster_engine/credentials/credential_vm.adoc [hunk 6]
```diff
@@ -106,3 +106,18 @@ You can create a cluster that uses this credential by completing the steps in xr
...
+    namespace: <managed-cluster-namespace>
+type: Opaque
+data:
```
Comment (manosnoam) [inline multicluster_engine/credentials/credential_vm.adoc]: The data part is missing, it is unique to VMware platform.

**Review by manosnoam — CHANGES_REQUESTED** [review]

File: multicluster_engine/credentials/credential_aws.adoc (hunk 2 above)
Comment (oafischer) [reply multicluster_engine/credentials/credential_aws.adoc]: Fixed

File: multicluster_engine/credentials/credential_azure.adoc (hunk 3 above)
Comment (oafischer) [reply multicluster_engine/credentials/credential_azure.adoc]: Fixed

File: multicluster_engine/credentials/credential_google.adoc (hunk 4 above)
Comment (oafischer) [reply multicluster_engine/credentials/credential_google.adoc]: Fixed

File: multicluster_engine/credentials/credential_openstack.adoc (hunk 5 above)
Comment (oafischer) [reply multicluster_engine/credentials/credential_openstack.adoc]: Fixed

File: multicluster_engine/credentials/credential_vm.adoc (hunk 6 above)
Comment (oafischer) [reply multicluster_engine/credentials/credential_vm.adoc]: Fixed

File: multicluster_engine/credentials/credential_azure.adoc (hunk 3 above)
Comment (manosnoam) [reply multicluster_engine/credentials/credential_azure.adoc]: Each platform has the data part **different**.

For example, on Azure I use:
```
data:
      baseDomainResourceGroupName: $(echo -n "${azure_resource_group_name}" | base64 -w0)
      osServicePrincipal.json: $(base64 -w0 "${AZURE_CRED_JSON}")

```
Where azure_resource_group_name is the relevant one for your cluster, as found in RG list, generated by command `az group show`.

Usually it is:
`azure_resource_group_name=${ocp_infra_id}-rg`

Where ocp_infra_id is retreived by:
`oc get -o jsonpath='{.status.infrastructureName}{"\n"}' infrastructure cluster`

File: multicluster_engine/credentials/credential_google.adoc (hunk 4 above)
Comment (manosnoam) [reply multicluster_engine/credentials/credential_google.adoc]: Same goes for GCP - the data is unique:

```
data:
        osServiceAccount.json: $(base64 -w0 "${GCP_CRED_JSON}")
```

File: multicluster_engine/credentials/credential_openstack.adoc (hunk 5 above)
Comment (manosnoam) [reply multicluster_engine/credentials/credential_openstack.adoc]: Same goes for Openstack - the data is unique:

```
data:
                clouds.yaml: $(base64 -w0 "${OSP_CRED_YAML}")
                cloud: $(echo -n "openstack" | base64 -w0)
```

File: multicluster_engine/credentials/credential_vm.adoc (hunk 1 above)
Comment (manosnoam) [reply multicluster_engine/credentials/credential_vm.adoc]: I'm quite sure for VMware the data is also unique.
Please check with CLC team.

**Review by manosnoam — CHANGES_REQUESTED** [review]

File: multicluster_engine/credentials/credential_vm.adoc (hunk 1 above)
Comment (oafischer) [reply multicluster_engine/credentials/credential_vm.adoc]: @elgnay

**Review by cadawson — APPROVED** [review]
This looks good to me. I will reach out to Le tonight to see if he or someone else can close out the open questions. Thanks!

File: multicluster_engine/credentials/credential_vm.adoc (hunk 1 above)
Comment (elgnay) [reply multicluster_engine/credentials/credential_vm.adoc]:
```
data:
      username: $(echo -n "${VMW_USERNAME}" | base64 -w0)
      password.json: $(base64 -w0 "${VMW_PASSWORD}")
```

**Comment by elgnay** [comment]
I noticed that the guide in `Managing a credential by using the console` is to create user credential (secret A) in user namespace which can be used by ACM console to provision cluster later, while the guide in `Creating an opaque secret by using the API` is to create secret (secret B) in cluster namespace which will be used by Hive to provision cluster. Normally secret A contains more information than Secret B. When provision cluster with ACM console, Secret A will be copied to the cluster namespace as Secret B. Please make sure the reader know the difference.

**Comment by oafischer** [comment]
Made some fixes, is this ready to merge @manosnoam?

**Comment by oafischer** [comment]
I assume this is fine, merging. If we need more updates we can open another issue.

**Pull request merged**